defaults.js

```diff
@@ -13,19 +15,32 @@ var exists = require('./exists');
  */
 module.exports = defaults;
 
-function defaults (target, source) {
+function defaults (target, source, deep) {
   if (arguments.length === 1) {
     source = target;
     return function (target) {
       return defaults(target, source);
     };
+  } else if (arguments.length === 2 && isBoolean(source)) {
```

Comment:
Using arguments length here is good, but can be limiting. 
I think we can be more flexible with this case. I would check `isBoolean(source)` alone 
and not worry about the arguments length. It's more flexible just in case the user is using
defaults in combination with another function that happens to call this function with the right
arguments (but maybe a few more - that we can ignore).